library/choose_disk.py

```diff
@@ -0,0 +1,651 @@
+                if infinite is False:
+                    break
+            elif match_count > 0:
+                # We only found a subset of the required features
+                logger.info("  %50s partially matched with %d/%d items", physical_disk, match_count, len(current_lookup))
+            else:
+                logger.info("  %50s no devices matched", physical_disk)
+
+    # Let's prepare the final output
+    final_disks = {}
+
+    # Matching devices are named base on the user-provided name + incremental number
+    # If we look for {'storage_disks': {'count': 2, 'ceph_type': 'data', 'vendor': '0x1af4', 'size': 'gte(20 GB)'}}
+    # - first device will be named  : storage_disks_000
+    # - second device will be named : storage_disks_001
+    for matched_device in matched_devices:
+        for n in range(0, len(matched_devices[matched_device])):
+            name = matched_device
+            if len(matched_devices[matched_device]) > 1:
+                name = "%s_%03d" % (matched_device, n)
+            final_disks[name] = matched_devices[matched_device][n]
+
+    return final_disks
+
+
+def expand_disks(lookup_disks, ceph_type="", module=None):
+    '''
+    Read the disks structure and expand them according to the count directive
+    '''
+    final_disks = {}
+
+    for disk in lookup_disks:
+        infinite = False
+        count = 0
+        if ceph_type:
+            # When legacy is enabled, let's enforce the count & type
+            count = 1
+            lookup_disks[disk]['ceph_type'] = ceph_type
+        else:
+            if 'count' not in lookup_disks[disk]:
+                fatal("disk '%s' should have a 'count' value defined" % disk, module)
+            if 'ceph_type' not in lookup_disks[disk]:
+                fatal("disk '%s' should have a 'ceph_type' value defined : {data | journal}" % disk, module)
+            if lookup_disks[disk]['ceph_type'] not in ['data', 'journal']:
+                fatal("disk '%s' doesn't have a valid 'ceph_type' defined, it should be : {data | journal}" % disk, module)
+            if 'count' in lookup_disks[disk]:
+                count = lookup_disks[disk]['count']
+                del lookup_disks[disk]['count']
+
+        #  If the count field is set to '*', let's consider an infinite number of devices to match
+        if '*' in str(count).strip():
+            infinite = True
+            count = 1
+
+        for n in range(0, int(count), 1):
+            final_disks["%s_%03d" % (disk, n)] = lookup_disks[disk]
+            if infinite is True:
+                final_disks["%s_%03d" % (disk, n)]["infinite"] = "1"
+
+    return final_disks
+
+
+def is_ceph_disk(partition):
+    '''
+    Check if a partition is containing some ceph structures
+    '''
+    stdout = subprocess.check_output(["lsblk", "-no", "PARTLABEL", "%s" % partition])
+
+    if "ceph data" in stdout:
+        return "data"
+
+    if "ceph journal" in stdout:
+        return "journal"
+
+    return ""
```

Comment:
the naming (and docstring) of this function is misleading since it implies a boolean response (vs. a string describing what this is). I would expect to have the return values changed or the function signature to be updated